system-tests/lib/cre/contracts/contracts.go

```diff
@@ -62,6 +63,20 @@ type dons struct {
 	c map[string]donConfig
 }
 
+func (d *dons) donsOrderedByID() []donConfig {
+	out := make([]donConfig, 0, len(d.c))
+	for _, don := range d.c {
+		out = append(out, don)
+	}
+
+	// Use sort library to sort by ID
+	sort.Slice(out, func(i, j int) bool {
+		return out[i].id < out[j].id
+	})
+
+	return out
```

Comment:
Er wait even cleaner (and more efficient) with newer seq funcs:
```suggestion
	return slices.SortedFunc(maps.Values(d.c), func(a, b donConfig) bool { return a.id < b.id })
```